src/sources/file.rs

```diff
@@ -532,8 +541,15 @@ pub fn file_source(
         let span2 = span.clone();
         let mut messages = messages.map(move |line| {
             let _enter = span2.enter();
-            let mut event =
-                create_event(line.text, &line.filename, &host_key, &hostname, &file_key);
+            let mut event = create_event(
+                line.text,
+                &line.filename,
+                line.offset,
+                &host_key,
+                &hostname,
+                &file_key,
+                &offset_key,
+            );
```

Comment:
This calling convention is getting unwieldy, particularly with the interleaving of the source fields and the output key names. I'm actually surprised clippy didn't complain about it. Consider adding all the keys to a new struct and pass that in.